core/iwasm/interpreter/wasm_runtime.c

```diff
@@ -2467,7 +2467,8 @@ wasm_instantiate(WASMModule *module, WASMModuleInstance *parent,
             goto fail;
         }
         for (i = 0; i < module->table_seg_count; i++) {
-            if (wasm_elem_is_active(module->table_segments[i].mode))
+            if (wasm_elem_is_active(module->table_segments[i].mode)
+                || wasm_elem_is_declarative(module->table_segments[i].mode))
```

Comment:
Should also apply the change to aot_runtime.c?
https://github.com/bytecodealliance/wasm-micro-runtime/blob/62aca1727908848c194dfe52c7d62397f5df8441/core/iwasm/aot/aot_runtime.c#L1907-L1910